src/routes/Reserve.jsx

```diff
@@ -1,59 +1,98 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import BASE_URL from '../redux/constants';
+import { getVideogames } from '../redux/slices/videogamesSlice';
+import { selectJWT, selectVideogames } from '../redux/store';
 import '../styles/Login.scss';
-import { useNavigate } from 'react-router-dom';
 import '../styles/reserve.scss';
-import { useSelector } from 'react-redux';
-import { useState } from 'react';
 import getTomorrowDate from '../utils/getTomorrowDate';
 
 const Reserve = () => {
+  const dispatch = useDispatch();
   const navigate = useNavigate();
-  const mockData = useSelector((state) => state.videogames.all);
+  const selectData = useSelector(selectVideogames);
+  const jwt = useSelector(selectJWT);
+  const [searchParams] = useSearchParams();
+  const initialID = searchParams.get('videogameId') ?? 0;
   const [days, setDays] = useState(0);
-  const [selectedVideogameId, setSelectedVideogameId] = useState(0);
+  const [selectedVideogameId, setSelectedVideogameId] = useState(initialID);
+
+  useEffect(() => {
+    dispatch(getVideogames());
+  }, [dispatch]);
+
+  if (!selectData) {
+    return <h1>Loading...</h1>;
+  }
```

Comment:
I would suggest implement the current Spinner component instead of this Loading title.
```shh
<div className="d-flex justify-content-center align-items-center w-100 h-100">
    <Spinner />
</div>
```